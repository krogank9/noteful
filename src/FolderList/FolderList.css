.folderList {
	list-style: none;
	padding: 0;
	margin: 0;
}

.folderButton {
	display: block;
	width: 100%;
	padding: 8px;
}

.folderButton.active {
	background-color: blue;
	color: white;
}
