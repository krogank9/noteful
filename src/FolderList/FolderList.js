import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { withRouter } from 'react-router';

import './FolderList.css';

import NotefulContext from '../NotefulContext.js';

class FolderList extends React.Component {
	
	static contextType = NotefulContext;
	
	render() {
		return (
			<ul className="folderList">
				{this.context.folders.map(folder => {
					return (
						<li key={folder.id}>
							<NavLink className="folderButton" to={"/folder/"+folder.id}>
								{folder.name}
							</NavLink>
						</li>
					);
				})}
			</ul>
		);
	}
}

export default FolderList;
