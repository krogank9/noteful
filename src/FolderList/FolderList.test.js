import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FolderList from './FolderList'

describe(`FolderList component`, () => {
	const props = {
		"folders": [
			{
				"id": "1",
				"name": "folder A"
			},
			{
				"id": "2",
				"name": "folder B"
			},
			{
				"id": "3",
				"name": "folder C"
			}
		],
	}

	it('renders with no props', () => {
		const wrapper = shallow(<FolderList />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders the FolderList given props', () => {
		const wrapper = shallow(<FolderList {...props} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
