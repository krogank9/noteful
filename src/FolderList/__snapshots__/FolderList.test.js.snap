// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderList component renders the FolderList given props 1`] = `
<ul
  className="folderList"
>
  <li
    key="1"
  >
    <NavLink
      className="folderButton"
      to="/folder/1"
    >
      folder A
    </NavLink>
  </li>
  <li
    key="2"
  >
    <NavLink
      className="folderButton"
      to="/folder/2"
    >
      folder B
    </NavLink>
  </li>
  <li
    key="3"
  >
    <NavLink
      className="folderButton"
      to="/folder/3"
    >
      folder C
    </NavLink>
  </li>
</ul>
`;

exports[`FolderList component renders with no props 1`] = `
<ul
  className="folderList"
/>
`;
